### PR TITLE
samples: bluetooth: mesh: Fix mesh samples overlays

### PR DESCRIPTION
**Pull request by PavelVPV**
Fix mesh samples overlays for bbc_microbit introduced in https://github.com/zephyrproject-rtos/zephyr/pull/55927.

### CODE REVIEW

File: samples/bluetooth/mesh/prj.conf [hunk 1]
```diff
@@ -39,7 +39,16 @@ CONFIG_BT_MESH_APP_KEY_COUNT=2
...
+
+#CONFIG_BT_DEBUG_LOG=y
+
+#CONFIG_BT_MESH_LOG_LEVEL_DBG=y
+#CONFIG_BT_MESH_NET_LOG_LEVEL_DBG=y
+#CONFIG_BT_MESH_TRANS_LOG_LEVEL_DBG=y
+#CONFIG_BT_MESH_LOG_LEVEL_DBG=y
+#CONFIG_BT_MESH_BEACON_LOG_LEVEL_DBG=y
+#CONFIG_BT_MESH_CRYPTO_LOG_LEVEL_DBG=y
+#CONFIG_BT_MESH_ADV_LOG_LEVEL_DBG=y
```
Comment (nordicjm) [inline samples/bluetooth/mesh/prj.conf:53]: No, if you want this, make a debug prj file

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: samples/bluetooth/mesh/boards/bbc_microbit.conf [hunk 2]
```diff
@@ -7,6 +7,8 @@ CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=1024
 CONFIG_MICROBIT_DISPLAY_STR_MAX=5
 CONFIG_GPIO=y
 
+CONFIG_HWINFO=n
```
Comment (nordicjm) [inline samples/bluetooth/mesh/boards/bbc_microbit.conf:10]: This was not in the original file, is it needed?

File: samples/bluetooth/mesh/boards/bbc_microbit.conf [hunk 3]
```diff
@@ -17,15 +19,31 @@ CONFIG_BT_BUF_ACL_RX_COUNT=3
 CONFIG_BT_BUF_EVT_DISCARDABLE_COUNT=3
 CONFIG_BT_EXT_ADV=n
 
+CONFIG_BT_CTLR_DUP_FILTER_LEN=0
+CONFIG_BT_OBSERVER=y
 CONFIG_BT_BROADCASTER=y
+CONFIG_BT_PERIPHERAL=n
 
+CONFIG_BT_CTLR_LE_ENC=n
 CONFIG_BT_CTLR_LE_PING=n
```
Comment (nordicjm) [inline samples/bluetooth/mesh/boards/bbc_microbit.conf:28]: What are you doing? None of this is needed, it is all part of the prj.conf file

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: samples/bluetooth/mesh/boards/bbc_microbit.conf (hunk 2 above)
Comment (nordicjm) [reply samples/bluetooth/mesh/boards/bbc_microbit.conf:10]: I see where it is, remove this and remove setting it from prj.conf and add it to an nrf52840dk board overlay instead

**Review by jhedberg — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by alxelax — APPROVED** [review]

**Pull request merged**